src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java

```diff
@@ -147,4 +157,34 @@ private void deduceDataClassificationForNewFields(CaseType caseType, CaseDetails
         jsonNode.ifPresent(value -> data.remove(CALLBACK_RESPONSE_KEY_STATE));
         return jsonNode.flatMap(value -> value.isTextual() ? Optional.of(value.textValue()) : Optional.empty());
     }
+
+    private void validateSignificantItem(CaseEvent eventTrigger, CallbackResponse callbackResponse) {
+        SignificantItem significantItem = callbackResponse.getSignificantItem();
+        List<String> errors = new ArrayList<>();
+
+        if (significantItem != null){
+            if(significantItem.getType() != ItemType.DOCUMENT){
+
+                errors.add("Significant Item type incorrect");
+            }
+            UrlValidator urlValidator = new UrlValidator();
+            if (!urlValidator.isValid(significantItem.getUrl())){
+                errors.add("Url Invalid");
+            }
+            if(isDescriptionEmptyOrNotWithInSpecifiedRange(significantItem)){
+               errors.add("Description should not be empty but also not more than 64 characters");
+            }
+            if(errors.size() == 0) {
+                eventTrigger.setSignificantDocument(true);
+                eventTrigger.setSignificantItem(significantItem);
```

Comment:
The event trigger is the definition of the event as per the definition, it shouldn't be used to pass the significant item around. Instead, the significant item should be directly returned by the function `invokeAboutToSubmitCallback()` to its caller.